src/components/auth/auth.service.js

```diff
@@ -0,0 +1,160 @@
+      member._id,
+    );
+
+    const userDetails = {
+      _id: member._id,
+      name: member.name,
+      email: member.email,
+      phone: member.phone,
+      address: member.address,
+      role: member.role,
+      membershipId: member.membershipId,
+      status: member.status,
+      createdAt: member.createdAt,
+    };
+
+    return {
+      userDetails,
+      accessToken: accessAndRefreshToken.accessToken,
+      refreshToken: accessAndRefreshToken.refreshToken,
+    };
+  } catch (err) {
+    const error = new Error(err.message);
+    throw error;
+  }
+};
+
+const loginUser = async (reqBody) => {
+  const { email, password } = reqBody;
+
+  try {
+    const userFound = await userMdl.user.findOne({ email });
+
+    if (!userFound) {
+      const error = new Error("USER_NOT_FOUND");
+      throw error;
+    }
+
+    if (!helper.decryptPassword(password, userFound.password)) {
+      const error = new Error("INVALID_PASSWORD");
+      throw error;
+    }
+
+    const accessAndRefreshToken = await helper.generateAccessAndRefreshToken(
+      userFound._id,
+    );
+    const userDetail = {
+      _id: userFound._id,
+      name: userFound.name,
+      email: userFound.email,
+      phone: userFound.phone,
+      address: userFound.address,
+      role: userFound.role,
+      membershipId: userFound.membershipId,
+      status: userFound.status,
+      createdAt: userFound.createdAt,
+    };
+    return {
+      userDetail,
+      accessToken: accessAndRefreshToken.accessToken,
+      refreshToken: accessAndRefreshToken.refreshToken,
+    };
+  } catch (err) {
+    const error = new Error(err.message);
+    throw error;
+  }
+};
+
+const logoutUser = async (req, res) => {
+  try {
+    const userFound = await userMdl.user.findById(req.user._id);
+    req.session.destroy((err) => {
+      if (err) {
+        const error = new Error(err.message);
+        throw error;
+      }
+    });
+    const blacklisted = await blacklistMdl.blacklist({
+      accessToken: req.accessToken,
+    });
+    await blacklisted.save();
+
+    userFound.refreshToken = null;
+    await userFound.save();
+
+    res.clearCookie("connect.sid");
+    return { message: "User logged out successfully" };
+  } catch (err) {
+    const error = new Error(err.message);
+    throw error;
+  }
+};
+
+const resetPassword = async (reqBody) => {
+  const { email, new_password, confirm_password } = reqBody;
+  try {
+    const userFound = await userMdl.user.findOne({ email });
+
+    if (!userFound) {
+      const error = new Error("USER_NOT_FOUND");
+      throw error;
+    }
+
+    if (helper.decryptPassword(new_password, userFound.password)) {
+      const error = new Error("CURRENT_PASSWORD");
+      throw error;
+    }
+    const hashedPassword = helper.encryptPassword(new_password);
+
+    if (!helper.decryptPassword(confirm_password, hashedPassword)) {
+      const error = new Error("PASSWORD_NOT_SAME");
```

Comment:
reset password tyare kare jyare password bhuli gyo hoi user. so you cant expect ke ae juno password ahiya aapse.

reset mate tyare ane email karvo pdse with some token. ae token thi pachi reset kari sakay password. RND kar, idea avse.

Reply:
ha ae old password vali error atle muki che k jo change krti vakhte unintentionally current password j nakhe to btavi dey k aa tmro current password j che jenathi reset-password ni process skip thy jay